doc/source/whatsnew/v1.5.0.rst

```diff
@@ -995,6 +995,7 @@ Reshaping
 - Bug in :func:`get_dummies` that selected object and categorical dtypes but not string (:issue:`44965`)
 - Bug in :meth:`DataFrame.align` when aligning a :class:`MultiIndex` to a :class:`Series` with another :class:`MultiIndex` (:issue:`46001`)
 - Bug in concatenation with ``IntegerDtype``, or ``FloatingDtype`` arrays where the resulting dtype did not mirror the behavior of the non-nullable dtypes (:issue:`46379`)
+- Bug in :func:`concat` losing dtype of columns when ``join="outer"`` and ``sort=True`` (:issue:`47329`)
```

Comment:
issue is marked as 1.4.4. ok to leave in 1.5 though (if so just change the issue and ignore this comment)

Reply:
Changed the milestones.

initially we though this happens only
for ea dtypes, but this was wrong. Occurs also for numpy dtypes